koku/masu/test/processor/test_expired_data_remover.py

```diff
@@ -136,3 +139,86 @@ def test_remove_items_only(self, mock_purge):
+
+        def insert_and_assert_delete_cost_usage_manifests(provider_type, provider_uuid):
+            """
+            Insert CostUsageReportManifests for a specific provider.
+
+            Assert that the expired records for that provider are deleted.
+            """
+            remover = ExpiredDataRemover(self.schema, provider_type)
+            expiration_date = remover._calculate_expiration_date()
+            this_month = datetime.today().replace(day=1)
+            day_before_cutoff = expiration_date - relativedelta.relativedelta(days=1)
+
+            # Record A
+            manifest_creation_datetime = this_month
+            manifest_updated_datetime = manifest_creation_datetime + relativedelta.relativedelta(days=2)
+            record_a_uuid = uuid4()
+            data = {
+                "assembly_id": record_a_uuid,
+                "manifest_creation_datetime": manifest_creation_datetime,
+                "manifest_updated_datetime": manifest_updated_datetime,
+                "billing_period_start_datetime": this_month,
+                "num_processed_files": 1,
+                "num_total_files": 1,
+                "provider_id": provider_uuid,
+            }
+            manifest_entry = CostUsageReportManifest(**data)
+            manifest_entry.save()
+
+            # Record B
+            # This record should be deleted because the billing_period_start_datetime
+            # is before the expiration_date
+            record_b_uuid = uuid4()
+            day_before_cutoff_data = {
+                "assembly_id": record_b_uuid,
+                "manifest_creation_datetime": manifest_creation_datetime,
+                "manifest_updated_datetime": manifest_updated_datetime,
+                "billing_period_start_datetime": day_before_cutoff,
+                "num_processed_files": 1,
+                "num_total_files": 1,
+                "provider_id": provider_uuid,
+            }
+            manifest_entry_2 = CostUsageReportManifest(**day_before_cutoff_data)
+            manifest_entry_2.save()
+
+            # Record C
+            # This record should not get deleted as it occurs on the date of the expiration, not before.
+            record_c_uuid = uuid4()
+            day_of_expiration_data = {
+                "assembly_id": record_c_uuid,
+                "manifest_creation_datetime": manifest_creation_datetime,
+                "manifest_updated_datetime": manifest_updated_datetime,
+                "billing_period_start_datetime": expiration_date,
+                "num_processed_files": 1,
+                "num_total_files": 1,
+                "provider_id": provider_uuid,
+            }
+            manifest_entry_3 = CostUsageReportManifest(**day_of_expiration_data)
+            manifest_entry_3.save()
+
+            remover.remove()
+            # Check if record A and C still exist. B should be deleted.
+            record_a = CostUsageReportManifest.objects.filter(assembly_id=record_a_uuid)
```

Comment:
I think you can do a `.count()` to return the number of rows that match this inside of doing the len(repord_a)

Reply:
@myersCody added.